src/Swarrot/Processor/Retry/RetryProcessor.php

```diff
@@ -61,7 +61,11 @@ public function process(Message $message, array $options)
                 $properties
             );
 
-            $key = str_replace('%attempt%', $attempts, $options['retry_key_pattern']);
+            $key = str_replace(
+                '%attempt%',
+                ($attempts > $options['retry_attempts']) ? $options['retry_attempts'] : $attempts,
```

Comment:
That would most certainly be possible and seems more logical indeed. What about an option that when with a specific routing key would use that key instead of `retry_key_pattern`? I can't think of a good name though. What about `retry_final_key`?
I think that `keep_retrying` would then become unnecessary.